CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+
+- Identify or create a GitHub Issue.
+- Create a topic branch from where to base the contribution. This is usually the main branch.
+- Make commits of logical units.
+- Make sure commit messages are in the proper format (see below).
+- Push changes in a topic branch to a personal fork of the repository.
+- Submit a pull request to the operator-framework/operator-controller repository.
+- Wait and respond to feedback from the maintainers listed in the OWNERS file.
+
+### Large Contributions
+
+The Operator Controller project is an open source project featuring contributors from many companies and backgrounds.
+In an effort to coordinate ongoing efforts from multiple contributors, large scale features are tracked on the
+[OLM V1 GitHub Project](https://github.com/orgs/operator-framework/projects/8/views/14?pane=info) in a `Milestone #` tab.
+Unassigned tickets in a `Milestone #` are available for contributors to take, simply assign the ticket to yourself and
+reach out to the community by commenting on the issue or by messaging us on the [#olm-dev](https://kubernetes.slack.com/archives/C0181L6JYQ2) Slack Channel.
+
+If you are interested in proposing a "focus" for a milestone, the contribution workflow should be similar to the following:
+
+- Create a [GitHub Issue](https://github.com/operator-framework/operator-controller/issues/new), labeled with "milestone-proposal", that includes a link to a [HackMD](https://hackmd.io) capturing the proposed changes.
+- Add an item to the [OLM Working Group and Issue Triage Meeting Agenda](https://docs.google.com/document/d/1Zuv-BoNFSwj10_zXPfaS9LWUQUCak2c8l48d0-AhpBw/edit) to introduce your design to the community.
+- Introduce your proposed changes at the weekly [OLM Working Group Meeting](https://github.com/operator-framework/community#operator-lifecycle-manager-working-group) 
+
+The community will then discuss the proposal, consider the benefits and cost of the feature, iterate on the design, and decide if it project should pursue the design. If accepted, the design will be queued up for a subsequent milestone.
+
+Thanks for contributing!
+
+### Code Review
+
+Contributing PRs with a reasonable title and description can go a long way with helping the PR through the review
+process.
+
+When opening PRs that are in a rough draft or WIP state, prefix the PR description with `WIP: ...` or create a draft PR.
+This can help save reviewer's time by communicating the state of a PR ahead of time. Draft/WIP PRs can be a good way to
+get early feedback from reviewers on the implementation, focusing less on smaller details, and more on the general
+approach of changes.
+
+When contributing changes that require a new dependency, check whether it's feasible to directly vendor that
+code [without introducing a new dependency](https://go-proverbs.github.io/).
+
+Currently, PRs required at least one approval from a operator-controller maintainer in order to get merged.
```

Comment:
Can we add a section mentioning requirements to open a PR. Something like [this](https://sdk.operatorframework.io/docs/contribution-guidelines/opening-pull-requests/#pr-checklist). Since this repository is in a fairly new state, we could ask for PR description, tests, relevant documentation as some tips for opening a PR.

Reply:
That's info that could be added to the PR template.

Reply:
I think this is covered by [the existing PR template](https://github.com/operator-framework/operator-controller/blob/main/.github/workflows/pull_request_template.md).

Reply:
@varshaprasad96 please mark this as resolved if you agree.